### PR TITLE
Feature/9 linter

### PR DESCRIPTION
**Pull request by anxolin**
- Add husky githook to lint ant test automatically before commiting and pushing.
- Fix lint errors/warnings
- Changes the rules for prettifier/linter to use 2 spaces instead of 4 for indentation

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr16--dexreact.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr16--dexreact.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr16--dexreact.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr16--dexreact.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr16--dexreact.review.gnosisdev.com

File: package.json [hunk 1]
```diff
@@ -6,10 +6,17 @@
...
+    "lint": "eslint src --ext .ts,.tsx,.js",
+    "lint:fix": "eslint --fix . --ext .ts,.tsx,.js",
     "stats": "NODE_ENV=production webpack --mode production --env.stats --profile --json > stats.json && webpack-bundle-analyzer ./stats.json ./dist",
-    "test": "echo Testing... && echo OK"
+    "test": "echo Testing... && echo OK",
+    "verify": "npm run lint && npm run test"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run verify",
```
Comment (fleupold) [inline package.json]: Do you need the same hook for commit and push?

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,10 @@ cache: 'yarn'
 script:
   - yarn test
   - yarn build
+before_install:
```
Comment (fleupold) [inline .travis.yml]: Is this PR introducing linting or fixing the travis script?

Comment (anxolin) [reply .travis.yml]: Yep, you are right, I was fixing the travis script in the wrong branch

File: package.json (hunk 1 above)
Comment (anxolin) [reply package.json]: I'll leave only the pre-commit for now

**Comment by fleupold** [comment]
@anxolin the gnosis info messages are awesome, but can we adjust the script to only post once per  PR (it should still build on every push, just not comment)? At the moment this guy is extremely spammy.

Worst case, the URL can be derived manually from the PR number.

Or maybe this guy can be a non-blocking github check that you usually find at the end of a PR.

**Comment by anxolin** [comment]
I'll bring in to conversation to Raul (@luarx ), father of the amazing PRaul functionality.

What would you say is the best way to do this part only one time?
image

**Comment by luarx** [comment]
Maybe that could be achieved if we check that there is already a comment which contains the PR preview URL before posting a new one. We can try it!

**Comment by fleupold** [comment]
>We can try it!

That would be cool. I was hoping travis would set a flag when it's the first time a PR is built, but I couldn't find anything.

**Comment by anxolin** [comment]
Yes I also looked for a env var, but didn't find anything relevant.

Also other approach is to create a `dummy file` in node_modules, that it's cached between compilations. Although I'm not 100% sure how the caching works, and if it's just caching between compilations of the same branch

```bash
IF dummy_file doesnt exist THEN
   touch dummy_file
   publish_comment()
ENDIF 
```

**Comment by anxolin** [comment]
I'm thinking, even if the `node_modules` would be reused between different branches, we can still use  the `PR number` in the name of the *dummy file*

**Comment by luarx** [comment]
We can try the @anxolin option also if the first one doesn't work as expected!

**Pull request merged**